kubetest2/pkg/types/types.go

```diff
@@ -79,6 +79,17 @@ type Deployer interface {
 	Build() error
 }
 
+// Some testers may need information about the deployer not available in the standard
+// deployer interface.
+
+// GinkgoableDeployer adds a few methods needed for ginkgo tests.
+type GinkgoableDeployer interface {
+	Deployer
+
+	Kubeconfig() (string, error)
+	Provider() string
```

Comment:
we should document defaulting, most will be defaulted

Reply:
I don't understand what you mean.  Just that most deployers will implement this method as returning a constant?

Reply:
imho deployers shouldn't need to implement this unless they need non-defaults. only legacy deployers specify this, there's no provision currently to add more providers to the e2e.test binary

Reply:
Moved to its own interface.